packages/polling/src/ratelimiter.ts

```diff
@@ -123,13 +123,72 @@ export class Debouncer<T = any, U = any> extends RateLimiter<T, U> {
  * @typeparam U - The rejected type of the underlying function. Defaults to any.
  */
 export class Throttler<T = any, U = any> extends RateLimiter<T, U> {
+  /**
+   * Instantiate a throttler.
+   *
+   * @param fn - The function being throttled.
+   *
+   * @param options - Throttling configuration or throttling limit in ms.
+   *
+   * #### Notes
+   * Both `leading` and `trailing` default to `true`; `limit` defaults to `500`.
+   * When both `leading` and `trailing` are `true`, `leading` has priority.`
+   */
+  constructor(fn: () => T | Promise<T>, options?: Throttler.IOptions | number) {
+    super(fn, typeof options === 'number' ? options : options && options.limit);
+    let leading = true;
+    let trailing = true;
+    if (typeof options !== 'number') {
+      options = options || {};
+      leading = 'leading' in options ? options.leading! : true;
+      trailing = 'trailing' in options ? options.trailing! : true;
```

Comment:
Is trailing flag required ? Can't it be considered as trailing when the leading is set as false ? With the current way of passing options, the throttler has to be instantiated as `new Throttler(fn, {leading:false, trailing:true}); `to ensure it'll be trailing. Just doing `new Throttler(fn, { trailing:true });  ` will still be considered as leading as by default leading is set to true which will override the options trailing flag.

Reply:
I think this probably makes sense, with the current implementation. I was wondering if *both* `leading` and `trailing` could be true, where we privilege the *leading call* and we wait until the *trailing edge* of the cycle. But given the underlying `RateLimiter` I don't see a straightforward way to implement that logic.

Reply:
Because if `leading` is `false` and `trailing` is `true`, it would mean that we invoke the *last call* before the *trailing edge* of the cycle, which requires storing invocations in a queue. And since invocations are basically meant to be replaceable with one another, I don't think this makes sense.

cc: @jasongrout